CMakePresets.json

```
@@ -0,0 +1,76 @@
+{
+    "version": 4,
+    "include": ["cmake/presets/ci.json"],
+    "configurePresets": [
+        {
+            "name": "base",
+            "hidden": true,
+            "binaryDir": "${sourceDir}/build-${presetName}"
```

My personal preference would be nesting the build folder and adding buildPresets. So that there won't be multiple `build-` folders in the `sourceDir` 
```suggestion
            "binaryDir": "${sourceDir}/build/${presetName}"
```

---

I would agree. Although typically there wouldn't be too many different build dirs anyway.

---

Thinking about it, I usually build in a `build/<branchName>` directory to keep different builds separated. But I guess that information is not available for CMakePresets.

---

No, but it's always possible to overwrite this from the CLI